src/Symfony/Component/Yaml/Parser.php

```diff
@@ -230,8 +230,12 @@ private function doParse($value, $flags)
                         $this->parseBlock($this->getRealCurrentLineNb() + 1, $this->getNextEmbedBlock(null, true), $flags)
                     );
                 } else {
-                    if (isset($values['leadspaces'])
-                        && self::preg_match('#^(?P<key>'.Inline::REGEX_QUOTED_STRING.'|[^ \'"\{\[].*?) *\:(\s+(?P<value>.+?))?\s*$#u', $this->trimTag($values['value']), $matches)
+                    if (
+                        isset($values['leadspaces'])
+                        && (
+                            '!' === $values['value'][0]
```

Comment:
Are we sure that `$values['value'][0]` is always set here?

Reply:
Yes, it's checked in the first part of the if statement above.